arcgisUM.py

```diff
@@ -88,39 +88,84 @@ def getArcGISGroupByTitle(arcGISAdmin, title):
     return None
 
 
-def addCanvasUsersToGroup(instructorLog, group, courseUsers):
-    """Add new users to the ArcGIS group.  """
-    groupNameAndID = util.formatNameAndID(group)
-    
+def addCanvasUsersToGroup(group, courseUsers, instructorLog):
+    """Add new users to a specific ArcGIS group"""
+
     logger.info("addCanvasUsersToGroup: enter")
+    groupNameAndID = util.formatNameAndID(group)
     
     if len(courseUsers) == 0:
         logger.info('No new users to add to ArcGIS Group {}'.format(groupNameAndID))
+        instructorLog += "No new users were added.\n\n"
         return instructorLog
 
     logger.info('Adding Canvas Users to ArcGIS Group {}: {}'.format(groupNameAndID, courseUsers))
     # ArcGIS usernames are U-M uniqnames with the ArcGIS organization name appended.
     arcGISFormatUsers = formatUsersNamesForArcGIS(courseUsers)
     logger.debug("addCanvasUsersToGroup: formatted: {}".format(arcGISFormatUsers))
-    
-    results = group.add_users(arcGISFormatUsers)
-    logger.debug("adding: results: {}".format(results))
 
-    usersNotAdded = results.get('notAdded')
-    """:type usersNotAdded: list"""
     usersCount = len(arcGISFormatUsers)
-    usersCount -= len(usersNotAdded) if usersNotAdded else 0
+    listsOfFormattedUsernames = util.splitListIntoSublists(arcGISFormatUsers, 20)
+    usersNotAdded = []
+
+    for listOfFormattedUsernames in listsOfFormattedUsernames:
+        try:
+            results = group.add_users(listOfFormattedUsernames)
+            logger.debug("adding: results: {}".format(results))
+            usersNotAdded += results.get('notAdded')
+        except RuntimeError as exception:
+            logger.error('Exception while removing users from ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+            return None
+
+    usersCount -= len(usersNotAdded)
     logger.debug("usersCount: {}".format(usersCount))
-    logger.debug("aCUTG: instructorLog 1: [{}]".format(instructorLog))
     instructorLog += 'Number of users added to group: [{}]\n\n'.format(usersCount)
-    logger.debug("aCUTG: instructorLog 2: [{}]".format(instructorLog))
     if usersNotAdded:
         logger.warning('Warning: Some or all users not added to ArcGIS group {}: {}'.format(groupNameAndID, usersNotAdded))
-        instructorLog += 'Users not in group (these users need ArcGIS accounts created for them):\n' + '\n'.join(['* ' + userNotAdded for userNotAdded in usersNotAdded]) + '\n\n' + 'ArcGIS group ID number:\n{}\n\n'.format(group.id)
```

Comment:
You might want the group ID number around for support/debugging purposes.

Reply:
With the group name and ID now added to the `instructorLog` in `updateGroupUsers` from `main.py`, it seemed redundant to have this here. Thoughts?